cmd/smith/app/app.go

```diff
@@ -207,10 +210,35 @@ func NewFromFlags(flagset *flag.FlagSet, arguments []string) (*App, error) {
 	flagset.IntVar(&a.Workers, "workers", 2, "Number of workers that handle events from informers")
 	flagset.StringVar(&a.Namespace, "namespace", meta_v1.NamespaceAll, "Namespace to use. All namespaces are used if empty string or omitted")
 	pprofAddr := flag.String("pprof-address", "", "Address for pprof to listen on")
+	printBundleSchema := flag.String("print-bundle-schema", "", "Print Bundle schema and exit (specify format: json or yaml)")
 	if err := flagset.Parse(arguments); err != nil {
 		return nil, err
 	}
 
+	switch *printBundleSchema {
+	case "json":
+		enc := json.NewEncoder(os.Stdout)
+		enc.SetIndent("", "  ")
+		err := enc.Encode(resources.BundleCrd().Spec.Validation.OpenAPIV3Schema)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to marshal Bundle schema")
+		}
+		return nil, context.Canceled
+	case "yaml":
+		data, err := yaml.Marshal(resources.BundleCrd().Spec.Validation.OpenAPIV3Schema)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to marshal Bundle schema")
+		}
+		_, err = os.Stdout.Write(data)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed write schema YAML to stdout")
+		}
+		return nil, context.Canceled
+	case "":
```

Comment:
do you actually need a special `case ""` there? isn't `default` enough?
could be some code style thing I guess.

Reply:
if the flag is not set, you should skip printing a bundle schema and run the application I suppose?
as far as I see currently the default value is `""` (empty string), and you will exit with error int that case

Reply:
ah nvm, Go `switch` works differently from other languages (https://tour.golang.org/flowcontrol/9): 
> Go only runs the selected case, not all the cases that follow

can you add some comment there like
```go
case "":
  // flag not set, do nothing
```

Reply:
Added a comment